langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/BedrockEmbeddingIT.java

```diff
@@ -83,4 +83,70 @@ void testBedrockTitanEmbeddingModelV2() {
         assertThat(embeddingModel.dimension()).isEqualTo(256);
     }
 
+    @Test
+    void testBedrockCohereEmbedEnglishTextV3() {
+
+        BedrockCohereEmbeddingModel embeddingModel = BedrockCohereEmbeddingModel
+                .builder()
+                .region(Region.US_EAST_1)
+                .maxRetries(1)
+                .model(BedrockCohereEmbeddingModel.Types.CohereEmbedEnglishTextV3.getValue())
+                .build();
+
+        assertThat(embeddingModel).isNotNull();
+
+        List<TextSegment> segments = Collections.singletonList(TextSegment.from("one"));
+
+        Response<List<Embedding>> response = embeddingModel.embedAll(segments);
+        assertThat(response).isNotNull();
+
+        List<Embedding> embeddings = response.content();
+        assertThat(embeddings).hasSize(1);
+
+        Embedding embedding = embeddings.get(0);
+        assertThat(embedding.vector()).hasSize(1024);
+
+        TokenUsage tokenUsage = response.tokenUsage();
+        assertThat(tokenUsage.inputTokenCount()).isEqualTo(0);
```

Comment:
Why is this 0? Shouldn't it be positive?

Reply:
It would in case that BedrockCohere includes "tokenCount" in response. But it doesn't.

Reply:
If there is no token usage information from Bedrock/Cohere, perharps we should not return `TokenUsage` at all? Providing 0 might be misleading

Reply:
I've investigated this further

In case of AWS Titan e.g. TitanEmbedTextV2 ("amazon.titan-embed-text-v2:0") response **has** information about tokenUsage.
One response example
`
{
  "embedding": [
    -0.0510961189866066,
    0.11482273787260056,
    ...
    -0.03215036541223526,
    -0.013778728432953358
  ],
  "embeddingsByType": {
    "float": [
      -0.0510961189866066,
      0.11482273787260056,
      ... 
      -0.03215036541223526,
      -0.013778728432953358
    ]
  },
  "inputTextTokenCount": 5
}
`

On the other hand, let's say that we use one of Cohere Embedding model ("cohere.embed-english-v3") response **doesn't** have info on token usage.
One response example
`
{
  "id": "8ac5e2e6-efeb-49ac-a608-83a0e98fca97",
  "texts": [
    "hello, world!"
  ],
  "embeddings": [
    [
      -0.04019165,
      -0.01914978,
      ...
      0.045135498,
      -0.018478394,
      0.01713562
    ]
  ],
  "response_type": "embeddings_floats"
}
`

Now if you take a look at interface `BedrockEmbeddingResponse.java`, there is a method signature
`int getInputTextTokenCount();`

Since it is interface and all other BedrockResponses must implement this interface, models that doesn't have such info has to have some implemention. To me sound logical that in that case we place 0.

Or in other case, let me suggest to remove this `int getInputTextTokenCount();` from interface and implement only for those responses that have it?

Reply:
@staleks what about changing the return type of `BedrockEmbeddingResponse.getInputTextTokenCount()` to `Optional<Integer>` to reflect that it is not always available?

Reply:
done.

Reply:
Please return `null` (`TokenUsage` == `null`) for cohere, returning 0 is misleading.

Reply:
Again tokenUsage is determined in `AbstractBedrockEmbeddingModel.java` ln:54-65

`
        int totalInputToken = 0;
        final List<Embedding> embeddings = new ArrayList<>();
        for (T response : responses) {
            embeddings.addAll(response.toEmbeddings());
            if (response.getInputTextTokenCount().isPresent()) {
                totalInputToken += response.getInputTextTokenCount().get();
            }
        }

        return Response.from(
                embeddings,
                new TokenUsage(totalInputToken));
`
and not in implementation class (Titan or Cohere).
What you suggest then?

Reply:
I would change `BedrockEmbeddingResponse` to have `TokenUsage tokenUsage()` instead of `int getInputTextTokenCount();`